frame/staking/src/mock.rs

```diff
@@ -424,24 +433,45 @@ impl ExtBuilder {
 
 		let mut stakers = vec![];
 		if self.has_stakers {
-			let stake_21 = if self.fair { 1000 } else { 2000 };
-			let stake_31 = if self.validator_pool { balance_factor * 1000 } else { 1 };
-			let status_41 = if self.validator_pool {
-				StakerStatus::<AccountId>::Validator
-			} else {
-				StakerStatus::<AccountId>::Idle
-			};
-			let nominated = if self.nominate { vec![11, 21] } else { vec![] };
 			stakers = vec![
-				// (stash, controller, staked_amount, status)
-				(11, 10, balance_factor * 1000, StakerStatus::<AccountId>::Validator),
-				(21, 20, stake_21, StakerStatus::<AccountId>::Validator),
-				(31, 30, stake_31, StakerStatus::<AccountId>::Validator),
-				(41, 40, balance_factor * 1000, status_41),
-				// nominator
-				(101, 100, balance_factor * 500, StakerStatus::<AccountId>::Nominator(nominated)),
+				// (stash, ctrl, stake, status)
+				// these two will be elected in the default test where we elect 2.
+				(11, 10, self.balance_factor * 1000, StakerStatus::<AccountId>::Validator),
+				(21, 20, self.balance_factor * 1000, StakerStatus::<AccountId>::Validator),
+				// a loser validator
+				(31, 30, self.balance_factor * 500, StakerStatus::<AccountId>::Validator),
+				// an idle validator
+				(41, 40, self.balance_factor * 1000, StakerStatus::<AccountId>::Idle),
 			];
+			// optionally add a nominator
+			if self.nominate {
+				stakers.push((
+					101,
+					100,
+					self.balance_factor * 500,
+					StakerStatus::<AccountId>::Nominator(vec![11, 21]),
+				))
+			}
+			// replace any of the status if needed.
+			self.status.into_iter().for_each(|(stash, status)| {
+				if let Some((_, _, _, ref mut prev_status)) =
+					stakers.iter_mut().find(|s| s.0 == stash)
```

Comment:
Should we just do an `.unwrap` on find here? I would assume probs a mistake if the stash is not present